### PR TITLE
docs(components): Add content guidelines to Banner

### PR DESCRIPTION
**Pull request by agglasaur**
## Motivations

https://jobber.atlassian.net/browse/JOB-27849

## Changes

Added content guidelines for how to use banner types, and clarification of when to use banners (notably reducing the usage of success banners as we want to use toasts in lieu of them for feedback in both Jobber online and mobile when appropriate)

### Added

- 

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by agglasaur** [comment]
@chris-at-jobber didnt include the Dos/Donts kind of table for each but i do have examples of that if thats useful to include for this kind of stuff.

**Comment by chris-at-jobber** [comment]
I don't know if this is just me, but I find I'm skimming over the bold instructions straight to the "regular" text, possibly because it feels like a subtitle of the section header:

image

Looking at the common usage, I wonder if we could borrow a page from the error banner itself and do something like

image

File: packages/components/src/Banner/Banner.mdx [hunk 1]
```diff
@@ -33,28 +33,80 @@ import { Banner } from "@jobber/components/Banner";
...
+    }}
+    dismissible={false}
+  >
+    Your client import is complete
+  </Banner>
 </Playground>
 
 ## Warning message
 
+**Use warning banners when something requires when an action could have unseen
```
Comment (chris-at-jobber) [inline packages/components/src/Banner/Banner.mdx:68]: Typo in this sentence

File: packages/components/src/Banner/Banner.mdx [hunk 2]
```diff
@@ -33,28 +33,80 @@ import { Banner } from "@jobber/components/Banner";
...
 </Playground>
 
 ## Error message
 
+**Use error banners when something requires immediate attention and blocks the
+user.**
+
+Errors should explain what happened, and summarize how many issues need to be
+addressed in a concise manner. Detailed information about each error is the role
+of the [input validation](input-validation) component.
```
Comment (chris-at-jobber) [inline packages/components/src/Banner/Banner.mdx:90]: I guess there aren't a ton of specifics in here, but it might be helpful to link out to the Product Vocabulary as a"for guidance on writing error messages, check this out" resource

File: packages/components/src/Banner/Banner.mdx [hunk 3]
```diff
@@ -66,23 +118,23 @@ within a `Banner`. If you require an action, use the `primaryActions` prop. The
 <Playground>
   <Content>
     <Banner
-      type="success"
+      type="notice"
       primaryAction={{
-        label: "Poke Me",
+        label: "View Plans",
         onClick: () => alert("💥💥SMASH💥💥")
```
Comment (chris-at-jobber) [inline packages/components/src/Banner/Banner.mdx:124]: I approve of the de-Marvelification of the sandbox examples - can we change the alert contents for this and line 134 as well? Maybe something like "Plans" and "Refreshing"...

File: packages/components/src/Banner/Banner.mdx (hunk 2 above)
Comment (chris-at-jobber) [reply packages/components/src/Banner/Banner.mdx:90]: I wonder if it's worth building the error-list formatting into the component itself... this is alright for now but it creates some awkward spacing since the `ul` has a margin-bottom assigned to it, that we could resolve by adding something like `bannerContent > ul { margin-bottom: 0 }`, but that still requires end users to write <ul><li></li></ul> HTML for every instance of their error banners. 

TLDR _this is fine_ but I think we can improve this at the component level

**Review by chris-at-jobber — CHANGES_REQUESTED** [review]
@agglasaur looks good, one grammar/typo change required™, but otherwise just some discussion points

**Comment by agglasaur** [comment]
@chris-at-jobber ready for another gander when you have a moment

**Review by chris-at-jobber — APPROVED** [review]
Looks Great To Me!
